plugin/pkg/scheduler/factory/factory.go

```diff
@@ -168,14 +169,39 @@ func (factory *ConfigFactory) createMinionLW() *cache.ListWatch {
 	}
 }
 
-// pollMinions lists all minions and returns an enumerator for cache.Poller.
+// pollMinions lists all minions and filter out unhealthy ones, then returns
+// an enumerator for cache.Poller.
 func (factory *ConfigFactory) pollMinions() (cache.Enumerator, error) {
-	list := &api.NodeList{}
-	err := factory.Client.Get().Resource("minions").Do().Into(list)
+	allNodes := &api.NodeList{}
+	err := factory.Client.Get().Resource("minions").Do().Into(allNodes)
 	if err != nil {
 		return nil, err
 	}
-	return &nodeEnumerator{list}, nil
+	nodes := &api.NodeList{
+		TypeMeta: allNodes.TypeMeta,
+		ListMeta: allNodes.ListMeta,
+	}
+	for _, node := range allNodes.Items {
+		conditionMap := make(map[api.NodeConditionKind]*api.NodeCondition)
+		for i := 0; i < len(node.Status.Conditions); i++ {
```

Comment:
You found it :)

```for i := range node.Status.Conditions {``` is slightly shorter, and still safe. :)

This is also safe:

```go
for i := range node.Status.Conditions {
  cond := node.Status.Conditions[i]
  conditionMap[cond.Kind] = &cond
}
```

Alternatively, a non-pointer map is probably fine here, too.